compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/TextFieldPointerModifier.uikit.kt

```diff
@@ -0,0 +1,211 @@
+    enabled: Boolean,
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping
+): Modifier = if (enabled) {
+    if (isInTouchMode) {
+        val selectionModifier = getSelectionModifier(manager)
+        val tapHandlerModifier = getTapHandlerModifier(
+            interactionSource,
+            state,
+            focusRequester,
+            readOnly,
+            offsetMapping,
+            manager
+        )
+        tapHandlerModifier
+            .then(selectionModifier)
+            .pointerHoverIcon(textPointerIcon)
+    } else {
+        this
+            .mouseDragGestureDetector(
+                observer = manager.mouseSelectionObserver,
+                enabled = enabled
+            )
+            .pointerHoverIcon(textPointerIcon)
+    }
+} else {
+    this
+}
+
+@OptIn(InternalFoundationTextApi::class)
+private fun getTapHandlerModifier(
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping,
+    manager: TextFieldSelectionManager
+) = Modifier.pointerInput(interactionSource) {
+    /*
+    We need to move tap recognizer here from selection modifier (as it is in common) because:
+    1) we need to handle triple tap
+    2) without rewriting, we have onDoubleTap call and onTap call, and onDoubleTap will execute before onTap.
+     */
+
+    detectRepeatingTapGestures(
+        onTap = { touchPointOffset ->
+            tapTextFieldToFocus(
+                state,
+                focusRequester,
+                !readOnly
+            )
+            if (state.hasFocus) {
+                if (state.handleState != HandleState.Selection) {
+                    state.layoutResult?.let { layoutResult ->
+                        TextFieldDelegate.setCursorOffset(
+                            touchPointOffset,
+                            layoutResult,
+                            state.processor,
+                            offsetMapping,
+                            state.onValueChange
+                        )
+                        // Won't enter cursor state when text is empty.
+                        if (state.textDelegate.text.isNotEmpty()) {
+                            state.handleState = HandleState.Cursor
+                        }
+                    }
+                } else {
+                    manager.deselect(touchPointOffset)
+                }
+            }
+        },
+        onDoubleTap = {
+            manager.doRepeatingTapSelection(it, SelectionAdjustment.Word)
+        },
+        onTripleTap = {
+            manager.doRepeatingTapSelection(it, SelectionAdjustment.Paragraph)
+        }
+    )
+}
+
+private fun getSelectionModifier(manager: TextFieldSelectionManager): Modifier {
+    val selectionModifier =
+        Modifier.pointerInput(Unit) {
```

Comment:
done
Am I correct that in that case change of the manager leads to cancel coroutine and restart it?
And there is no need to use `by rememberUpdatedState` like in previous comment?

Reply:
Yes, the coroutine restarts, and `rememberUpdatedState` isn't needed. Usually we choose `rememberUpdatedState` if the gesture in the coroutine can deal with new properties, or/and it is undesirable to interrupt the gesture .

I suggested restarting here, because that is what used in `Modifier.longPressDragGestureFilter` in the previous code. After looking again, `rememberUpdatedState` is more correct here, as we don't want drag to be interrupted.

We won't see interruption with the current code, because `manager` seems never changes:
```
val manager = remember { TextFieldSelectionManager(undoManager) }
```

But better to rewrite to `rememberUpdatedState` to avoid future bugs.